src/workerd/api/global-scope.c++

```diff
@@ -470,6 +471,21 @@ void ServiceWorkerGlobalScope::sendHibernatableWebSocketMessage(
   KJ_IF_SOME(h, exportedHandler) {
     KJ_IF_SOME(handler, h.webSocketMessage) {
       auto promise = handler(lock, kj::mv(websocket), kj::mv(message));
+
+      // If we have a maximum event duration timeout set, we should prevent the actor from running
+      // for more than the user selected duration.
+      KJ_IF_SOME(timeout, eventTimeout) {
+        if (timeout > 0) {
+          // We must have an actor if we are delivering an hibernatable web socket message.
+          auto& actor = KJ_ASSERT_NONNULL(IoContext::current().getActor());
+
+          promise = promise.exclusiveJoin(IoContext::current().afterLimitTimeout(timeout * kj::SECONDS)
+            .then([&actor]() {
+              actor.shutdown(2, JSG_KJ_EXCEPTION(DISCONNECTED, Error,
```

Comment:
I left a comment on the commit instead of the file :man_facepalming: and github makes it extremely hard to find... To reiterate:

Now that I see it in a callsite, I'm really not fond of this being an int. In code, 2 being passed to a method is really unclear, and if I hadn't already known that this refers to a capnp schema I would be totally lost...

For the time being, can we at least set some variable like
```
uint16_t evictionCode = 2; actor.shutdown(evictionCode, /*etc*/);
```

We don't need to address it here, but it would be nice if we could be more explicit with the reasonCodes.